### PR TITLE
Comments

### PR DESCRIPTION
**Pull request by richawadaskar**
- Logged in users can make comments on posts
- Logged in users can edit/delete their own comments
- Logged in users can view all the comments they've created at /comments, and access the posts associated with each comment through the "Show" link.

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -1,10 +1,17 @@
...
   # GET /comments.json
   def index
-    @comments = Comment.all
+    # if the current user exists, then
+    puts "Session user"
+    puts session[:user_id].class
+    if current_user
+      @comments = Comment.where(user_id: session[:user_id])
+    end
   end
```
Comment (lar-ryan-nick) [inline app/controllers/comments_controller.rb]: What happends if someone goes to this path while not logged in?

File: app/views/comments/index.html.erb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <p id="notice"><%= notice %></p>
 
-<h1>Comments</h1>
+<h1>Your Comments</h1>
```
Comment (lar-ryan-nick) [inline app/views/comments/index.html.erb]: Aren't we supposed to allow unauthenticated users to see all comments still?

Comment (richawadaskar) [reply app/views/comments/index.html.erb]: The /comments is just for logged in users to see all their comments. 
At least that's my interpretation of it. 
Unauthenticated users can still view comments made on posts by viewing a post

File: app/controllers/comments_controller.rb (hunk 1 above)
Comment (richawadaskar) [reply app/controllers/comments_controller.rb]: They don't see any comments.

**Pull request merged**